File: gc.py
import sys

def main():
    filename = sys.argv[1]
    BUF_SIZE = 1024

    try:
        with open(filename, 'rb') as f:
    
            while True:
                buf = f.read(BUF_SIZE)
    
                if len(buf) == 0:
                    break
    
                sys.stdout.buffer.write(buf)
    
    except FileNotFoundError:
        sys.stderr.write(f"File {filename} not found!")


if __name__ == '__main__':
    
    main()

File: gc.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>

#define BUF_SIZE 2048

int main(int argc, char *argv[]) {
    using namespace std;

    if (argc != 2) {
        cout << "Usage: " << argv[0] << " <filename>" << endl;
        exit(1);
    }

    ifstream inFile(string(argv[1]), ifstream::in);
    char buf[BUF_SIZE];

    while (inFile.good()) {
        inFile.read(buf, BUF_SIZE);
        cout << buf;
        if (inFile.gcount() < BUF_SIZE) break;
    }
    inFile.close();

    return 0;
}
